legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/main/java/org/finos/legend/engine/persistence/components/relational/h2/sqldom/schemaops/statements/CopyStatement.java

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.h2.sqldom.schemaops.statements;
+
+import org.finos.legend.engine.persistence.components.relational.sqldom.SqlDomException;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.expresssions.table.Table;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.statements.DMLStatement;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.statements.SelectStatement;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schemaops.values.Field;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.finos.legend.engine.persistence.components.relational.sqldom.common.Clause.INSERT_INTO;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.utils.SqlGenUtils.CLOSING_PARENTHESIS;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.utils.SqlGenUtils.COMMA;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.utils.SqlGenUtils.OPEN_PARENTHESIS;
+import static org.finos.legend.engine.persistence.components.relational.sqldom.utils.SqlGenUtils.WHITE_SPACE;
+
+public class CopyStatement implements DMLStatement
+{
+    private Table table;
+    private final List<Field> columns;
+    private SelectStatement selectStatement;
+
+    public CopyStatement()
+    {
+        columns = new ArrayList<>();
+    }
+
+    public CopyStatement(Table table, List<Field> columns, SelectStatement selectStatement)
+    {
+        this.table = table;
+        this.columns = columns;
+        this.selectStatement = selectStatement;
+    }
+
+    /*
+     Copy GENERIC PLAN for H2:
+        INSERT INTO table_name (COLUMN_LIST)
+        SELECT [CONVERT("column_name", column_type) , ...]
+        FROM CSVREAD('{FILE_PATH}','{CSV_COLUMN_NAMES}',NULL)
```

Comment:
rather than NULL, we should model is as CSV Options
http://www.h2database.com/html/functions.html#csvread

Reply:
Sure, updated the comment - note that copy command will still use NULL for csv options. The underlying CSVREAD operation does support taking in different csv options.